doc/source/visualizing/plots.rst

```diff
@@ -920,52 +923,121 @@ Use any of the colormaps listed in the :ref:`colormaps` section.
-choose the smallest non-zero value in a dataset to be the ``linthresh`` value.
-As an example,
+Units can be left out, in which case they implicitly match the current display
+units of the colorbar (controlled with the ``set_unit`` method, see
+:ref:`_set-image-units`).
+
+Both ``zmin`` and ``zmax`` are optional, but note that they respectively default
+to 0 and 1, which can be widely inappropriate, so it is recommended to specify both.
+
+:meth:`~yt.visualization.plot_container.ImagePlotContainer.set_log` takes a boolean argument
+to select log (``True``) or linear (``False``) scalings.
+
+.. python-script::
+
+   import yt
+
+   ds = yt.load("IsolatedGalaxy/galaxy0030/galaxy0030")
+   slc = yt.SlicePlot(ds, "z", ("gas", "density"), width=(10, "kpc"))
+   slc.set_log(("gas", "density"), False) # switch to linear scaling
+   slc.save()
+
+One can switch to `symlog
+<https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.SymLogNorm.html?highlight=symlog#matplotlib.colors.SymLogNorm>`_
+by providing a "linear threshold" (``linthresh``) value.
+With ``linthresh="auto"`` yt will switch to symlog norm and guess an appropriate value
+automatically. Specifically the minimum absolute value in the image is used
+unless it's zero, in which case yt uses 1/1000 of the maximum value.
+
+.. python-script::
+
+   import yt
+
+   ds = yt.load_sample("IsolatedGalaxy")
+   slc = yt.SlicePlot(ds, "z", ("gas", "density"), width=(10, "kpc"))
+   slc.set_log(("gas", "density"), linthresh="auto")
+   slc.save()
+
+
+In some cases, you might find that the automatically selected linear threshold is not
+really suited to your dataset, for instance
 
 .. python-script::
 
    import yt
 
    ds = yt.load_sample("FIRE_M12i_ref11")
    p = yt.ProjectionPlot(ds, "x", ("gas", "density"))
-   p.set_log(("gas", "density"), True, symlog_auto=True)
+   p.set_log(("gas", "density"), linthresh="auto")
    p.save()
 
-Symlog is very versatile, and will work with positive or negative dataset ranges.
-Here is an example using symlog scaling to plot a positive field with a linear range of
-``(0, linthresh)``.
+An explicit value can be passed instead
 
 .. python-script::
 
    import yt
 
+   ds = yt.load_sample("FIRE_M12i_ref11")
+   p = yt.ProjectionPlot(ds, "x", ("gas", "density"))
+   p.set_log(("gas", "density"), linthresh=(1e-22, "g/cm**2"))
+   p.save()
+
+Similar to the ``zmin`` and ``zmax`` arguments of the ``set_zlim`` method, units
+can be left out in ``linthresh``.
+
+
+**Arbitrary norms**
+
+Alternatively, arbitrary matplotlib norms can be passed via the
+:meth:`~yt.visualization.plot_container.PlotContainer.set_norm` method. In that
+case, any numeric value is treated as having implicit units, matching the
+current display units. This alternative interface is more flexible, but
+considered experimental as of yt 4.1. Don't forget that with great power comes
+great responsibility.
+
+
+.. python-script::
+
+   import yt
+   from matplotlib.colors import TwoSlopeNorm
```

Comment:
The intention here was to demonstrate how to use norms that are part of the `matplotlib` api other that linear, log and symlog, since it should never be needed to setup these 3 manually.
And indeed, `TwoSlopeNorm` is linear-only. I guess in principle it should be possible to extend matplotlib to define the log equivalent but no such thing is shipped with matplotlib itself atm.

Reply:
ok, fair enough, happy to leave this then.